go-controller/pkg/factory/factory.go

```diff
@@ -329,28 +329,52 @@ func getObjectMeta(objType reflect.Type, obj interface{}) (*metav1.ObjectMeta, e
 	return nil, fmt.Errorf("cannot get ObjectMeta from type %v", objType)
 }
 
-func (wf *WatchFactory) addHandler(objType reflect.Type, namespace string, sel labels.Selector, funcs cache.ResourceEventHandler, processExisting func([]interface{})) *Handler {
+func getServiceSpec(objType reflect.Type, obj interface{}) (*kapi.ServiceSpec, error) {
```

Comment:
we don´t need this here, you can filter directly in the consumer of this handler, also there is no benefit in doing it here

This code is already quite complex, I suggest to avoid adding more things on top

Reply:
Sounds good! From the sounds of it I will be refactoring the handler a bit based on the comments regardless